todo_app/templates/index.html

```diff
@@ -27,28 +27,127 @@ <h2>To Do List</h2>
+                  <input type="hidden" name="toggle_item_id" value="{{ item.id }}">
+                  <input type="button" id="toggle_item_button_{{ item.id}}" class="btn btn-primary btn-sm" value="Update" onclick="if(document.getElementById('new_trello_list_id_{{ item.id }}').options[document.getElementById('new_trello_list_id_{{ item.id }}').selectedIndex].value == 'default'){window.alert('You must select a status to change to!')}else{document.getElementById('toggle_item_{{ item.id }}').submit()}">                
+              </form>
+
+            <form id="archive_item_{{ item.id }}" action="/archive_item" method="POST" class="form-inline">
+                <input type="hidden" name="archive_item_id" value="{{ item.id }}">
+                 <input type="submit" value="Archive" class="btn btn-link btn-sm mx-2">
+            </form>
+          </td>
+        </tr>
+        {% endfor %}
+      </tbody>
+      </table>
+
+      <h3>Completed Items</h3>
+      {% if view_model.done_items|length >= 5 %}
+      <br />
+      <div><button class="btn btn-primary btn-sm" type="button" data-toggle="collapse" data-target="#completedOlder" aria-controls="completedOlder">Show/Hide Older Tasks</button></div>
+      <br />
+      {% endif %}
+      <table class="table table-hover table-sm">
+        <thead>
+          <tr>
+            <th scope="col"><strong>Task</strong></th>
+            <th scope="col"><strong>Description</strong></th>
+            <th scopt="col"><strong>Due Date</strong></th>
+            {% if request.values.get('sort') == None %}
+            <th scope="col"><strong><a href="/?sort=2" class="table">Status</a></strong></th>
+            {% elif request.values.get('sort') == '2' %}
+            <th scope="col"><strong><a href="/?sort=1" class="table">Status</a> &circ;</strong></th>
+            {% elif request.values.get('sort') == '1' %}
+            <th scope="col"><strong><a href="/?sort=2" class="table">Status</a> &caron;</strong></th>
             {% endif %}
+            <th scope="col"><strong>Actions</strong></td>
+          </tr>
+      </thead>
+      <tbody>
+        {% for item in view_model.done_items %}
+        {% if view_model.done_items|length >= 5 %}
+          {% if item.get_modified_user_facing_date() == todays_date %}
+          <tr scope="row" class="collapse show" id="completedToday">
+          {% else %}
+          <tr scope="row" class="collapse" id="completedOlder">
+          {% endif %}
+        {% else %}
+          <tr scope="row">
+        {% endif %}
+          <td>{{ item.name }}</td>
+          <td>{{ item.description }}</td>
+          <td>{{ item.get_user_facing_date() }}</td>
+          <td id="Done_{{ item.id }}">Done</td>
           <td class="form-inline">
-              <form id="toggle_item_{{ item.id }}" action="/toggle_status" method="POST" class="form-inline">
+              <form id="toggle_item_form_{{ item.id }}" action="/toggle_status" method="POST" class="form-inline">
```

Comment:
You've renamed your form's id, but haven't caught everywhere it was referenced - e.g. in the Update buttons `onclick` method, you end with `document.getElementById('toggle_item_{{ item.id }}').submit()` which now errors; we should fix that up for the three update buttons (and this might be a good chance to think about how we could refactor that, so we'd only need to fix it once in future!)

Sadly this isn't caught by your end-to-end test because that submits the forms directly rather than by explicitly clicking on the relevant button!